appinventor/appengine/src/com/google/appinventor/server/storage/ObjectifyStorageIo.java

```diff
@@ -2025,6 +2025,14 @@ public ProjectSourceZip exportProjectSourceZip(final String userId, final long p
         @Override
         public void run(Objectify datastore) throws IOException {
           Key<ProjectData> projectKey = projectKey(projectId);
+          for (FileData fd: datastore.query(FileData.class).ancestor(projectKey)) {
+            String fileName = fd.fileName;
+            if (fileName.endsWith("scm") || fileName.endsWith("bky") || fileName.endsWith("yail")) {
```

Comment:
We should include the dot for the extension, otherwise I could upload an asset called Foobarscm and it would result in an IndexOutOfBoundsException because lastIndexOf will return -1 which would then be passed to substring. We should also check that the fileName starts with src/ for the  reason that we only want to apply this logic to actual source files and not inadvertently filter out assets.